src/Membrain/Memory.hs

```diff
@@ -39,6 +41,24 @@ newtype Memory (mem :: Nat) = Memory
     { unMemory :: Natural
     } deriving (Show, Read, Eq, Ord)
 
+instance Semigroup (Memory (mem :: Nat)) where
```

Comment:
Let's also implement `sconcat` and `stimes` methods of `Semigroup` for more efficiency